### PR TITLE
Feature: Allow to disable a plugin

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

Add the possibility to skip loading a plugin, including core plugins (but not limited to those).

**Why do we need this feature?**

An operator may want to disable a plugin.

**Who is this feature for?**

Grafana operators.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-operator-experience-squad/issues/614

**Special notes for your reviewer:**

Note that I have skipped adding a feature flag for this featue since it won't be used unless it's configured. Let me know if you still think a feature flag is necessary.

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by torkelo — APPROVED** [review]
I think this looks good

**Review by xnyo — APPROVED** [review]
LGTM! 🚀

**Review by wbrowne — COMMENTED** [review]
I think we could probably just do this within `pluginsintegration`, no? We can introduce the step there in here https://github.com/grafana/grafana/blob/main/pkg/services/pluginsintegration/pipeline/steps.go and then we also don't have to introduce a new config to the plugins config struct.

WDYT?

**Review by chri2547 — APPROVED** [review]
docs look good! Thanks for the contribution!

File: conf/defaults.ini [hunk 1]
```diff
@@ -1478,6 +1478,8 @@ public_key_retrieval_disabled = false
 # Force download of the public key for verifying plugin signature on startup. If disabled, the public key will be retrieved every 10 days.
 # Requires public_key_retrieval_disabled to be false to have any effect.
 public_key_retrieval_on_startup = false
+# Enter a comma-separated list of plugin identifiers to skip loading (including core plugins). These plugins will be hidden in the catalog.
+skip_plugins =
```
Comment (marefr) [inline conf/defaults.ini:1482]: Nit. Would perhaps suggest renaming it to  `disabled_plugins`. Something with "skip" that does feel weird to read for me without being able to pinpoint exactly why :smile:

**Review by marefr — APPROVED** [review]
LGTM with a nit

**Comment by andresmgot** [comment]
> I think we could probably just do this within pluginsintegration, no?

Right, I always forget :+1: 

> Nit. Would perhaps suggest renaming it to disabled_plugins

Sounds good to me :+1:

**Comment by andresmgot** [comment]
> and then we also don't have to introduce a new config to the plugins config struct.

@wbrowne as far as I can tell, the config change is still necessary because it's what's received in the pipeline but let me know if I am missing something.

**Comment by wbrowne** [comment]
> > and then we also don't have to introduce a new config to the plugins config struct.
> 
> @wbrowne as far as I can tell, the config change is still necessary because it's what's received in the pipeline but let me know if I am missing something.

Ah yeah I was thinking just also inject Grafana's `*setting.Cfg` there. I just worry that compared to some other config, someone might expect to set that and it automatically work (but won't because you need the right step). But also not a deal breaker!

**Pull request merged**